### PR TITLE
PKCS #11: implement identifying keys by label 

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -19881,6 +19885,76 @@ int PickHashSigAlgo(WOLFSSL* ssl, const byte* hashSigAlgo, word32 hashSigAlgoSz)
...
+            ret = wc_InitRsaKey_Label(rsaKey, (char*)data, heap, devId);
+        }
+        else if (id) {
+            ret = wc_InitRsaKey_Id(rsaKey, data, length, heap, devId);
+        }
+        if (ret == 0) {
+            *pkey = (void*)rsaKey;
+        }
+        else {
+            XFREE(rsaKey, heap, DYNAMIC_TYPE_EC);
```
Comment (fencekicker) [inline src/internal.c:19926]: Shouldn't this be DYNAMIC_TYPE_RSA?

File: src/internal.c [hunk 2]
```diff
@@ -19881,6 +19885,76 @@ int PickHashSigAlgo(WOLFSSL* ssl, const byte* hashSigAlgo, word32 hashSigAlgoSz)
...
+            ret = wc_ecc_init_label(ecKey, (char*)data, heap, devId);
+        }
+        else if (id) {
+            ret = wc_ecc_init_id(ecKey, data, length, heap, devId);
+        }
+        if (ret == 0) {
+            *pkey = (void*)ecKey;
+        }
+        else {
+            XFREE(ecKey, heap, DYNAMIC_TYPE_EC);
```
Comment (fencekicker) [inline src/internal.c:19949]: DYNAMIC_TYPE_ECC?

File: src/internal.c (hunk 1 above)
Comment (SparkiDev) [reply src/internal.c:19926]: Fixed. Thanks!

File: src/internal.c (hunk 2 above)
Comment (SparkiDev) [reply src/internal.c:19949]: Fixed. Thanks!

**Comment by SparkiDev** [comment]
New branch on which these changes have been applied to current master:
https://github.com/wolfSSL/wolfssl/pull/3523

This pull request closed and maintained for customer. They are using 4.3.0 so differences doesn't apply easily.

**Pull request closed without merging**